index.ts

```diff
@@ -113,32 +130,33 @@ function candidateToGeojson (candidate) {
  * @param {string} [$0.url]                       optional URL to override ESRI suggest endpoint
  * @return {Promise}                              A Promise that'll get resolved with the suggest result
  */
-export function autocomplete ({
+export function autocomplete({
+  boundary,
   clientId,
   clientSecret,
-  boundary,
   focusPoint,
   text,
   url
 }: BaseQuery & {
-  boundary?: Boundary,
-  focusPoint?: any,
-  text: string,
-}): Promise<BaseResponse> {
-  const geocoder = getGeocoder(clientId, clientSecret, url)
-  const options = {}
+  boundary?: Boundary
+  focusPoint?: Point
+  text?: string
+}): Promise<GeoJSON.FeatureCollection & { query: { text: string } }> {
+  const geocoder: GeocoderArcGIS = getGeocoder(clientId, clientSecret, url)
+  const options: { location?: string; searchExtent?: string } = {}
 
   if (focusPoint) {
-    options.location = lonlat.toString(focusPoint)
+    options.location = toString(focusPoint)
   }
 
   if (boundary) {
     options.searchExtent = boundaryToSearchExtent(boundary)
   }
 
   // make request to arcgis
-  return geocoder.suggest(text, options)
-    .then(response => {
+  return geocoder
+    .suggest(text, options)
+    .then((response: { suggestions: string[] }) => {
```

Comment:
Please use the recommended type in combination with the above comment:

```suggestion
    .then((response: { suggestions: SuggestOption[] }) => {
```

Reply:
Refactored in aecf758d679984328a465df451fd953bde17efbf